pkg/aws/eni/node.go

```diff
@@ -322,12 +322,13 @@ func (n *Node) findNextIndex(index int32) int32 {
 // CreateInterface without additional context embedded in order to make them
 // usable for metrics accounting purposes.
 const (
-	errUnableToDetermineLimits    = "unable to determine limits"
-	errUnableToGetSecurityGroups  = "unable to get security groups"
-	errUnableToCreateENI          = "unable to create ENI"
-	errUnableToAttachENI          = "unable to attach ENI"
-	errUnableToMarkENIForDeletion = "unable to mark ENI for deletion"
-	errUnableToFindSubnet         = "unable to find matching subnet"
+	errUnableToAttachENI                    = "unable to attach ENI"
+	errUnableToCreateENI                    = "unable to create ENI"
+	errUnableToDetermineFirstInterfaceIndex = "unable to determine first interface index"
+	errUnableToDetermineLimits              = "unable to determine limits, consider setting --update-ec2-adapter-limit-via-api=true"
```

Comment:
I think we should keep recommendations / mitigations to log msgs instead of error msgs. Errors should describe succinctly the cause of a problem, but it should not recommend any course of action. I believe it is cleaner if the caller of the function which errored to recommend a course of action / provide a mitigation instead.  This way the caller has the full context of the issue and can decide if it makes sense to log at that point in the code path.

Reply:
sounds good, I will revert this change then 👍, where exactly do you reckon we should be putting the advice?

Reply:
I think the first place that the user will encounter this error is in `pkg/nodediscovery.determineFirstInterfaceIndex()` so it would make sense to put it there to warn as early as possible

Reply:
sounds great, I added the warning at this place 👍